### PR TITLE
Merge AzureStorageService and AzureStorageServiceImpl and clean up tests

### PR DESCRIPTION
**Pull request by tlrx**
This pull request merges the AzureStorageService interface and the AzureStorageServiceImpl classes into one single AzureStorageService class. It also removes some tests in the repository-azure plugin that have not been executed for 2+ years.

### CODE REVIEW

File: plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageService.java [hunk 1]
```diff
@@ -19,27 +19,59 @@
...
-/**
- * Azure Storage Service interface
- * @see AzureStorageServiceImpl for Azure REST API implementation
- */
-public interface AzureStorageService {
+import static java.util.Collections.emptyMap;
+
+public class AzureStorageService extends AbstractComponent {
+
+    public static ByteSizeValue MIN_CHUNK_SIZE = new ByteSizeValue(1, ByteSizeUnit.BYTES);
```
Comment (vladimirdolzhenko) [inline plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageService.java]: should it be `final` ? seems to be a const semantics

**Review by vladimirdolzhenko — COMMENTED** [review]
I left a small comment

File: plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageService.java (hunk 1 above)
Comment (tlrx) [reply plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageService.java]: Right

**Comment by vladimirdolzhenko** [comment]
LGTM

**Comment by tlrx** [comment]
Thanks @vladimirdolzhenko

**Pull request merged**